website/client/components/guilds/discovery.vue

```diff
@@ -0,0 +1,98 @@
+<template lang="pug">
+.row
+  sidebar(@search="updateSearch", @filter="updateFilters")
+
+  .col-10
+    .clearfix
+        h1.page-header.float-left(v-once) {{ $t('publicGuilds') }}
+        b-form-select.float-right.sort-select(v-model='sort', :options='sortOptions')
+    .col-md-12
+      public-guild-item(v-for="guild in filteredGuilds", :key='guild._id', :guild="guild", :display-leave='true')
+      mugen-scroll(
+        :handler="fetchGuilds",
+        :should-handle="!loading && !this.hasLoadedAllGuilds",
+        :handle-on-mount="false",
+        v-show="loading",
+      )
+        span(v-once) {{ $t('loading') }}
+</template>
+
+<style>
+.sort-select {
+  margin: 2em;
+}
+</style>
+
+<script>
+import MugenScroll from 'vue-mugen-scroll';
+import PublicGuildItem from './publicGuildItem';
+import Sidebar from './sidebar';
+import groupUtilities from 'client/mixins/groupsUtilities';
+import bFormSelect from 'bootstrap-vue/lib/components/form-select';
+
+export default {
+  mixins: [groupUtilities],
+  components: { PublicGuildItem, MugenScroll, Sidebar, bFormSelect },
+  data () {
+    return {
+      loading: false,
+      hasLoadedAllGuilds: false,
+      lastPageLoaded: 0,
+      search: '',
+      filters: {},
+      sort: 'none',
+      sortOptions: [
+        {
+          text: 'None',
+          value: 'none',
+        },
+        {
+          text: 'Member Count',
+          value: 'member_count',
+        },
+        {
+          text: 'Recent Activity',
+          value: 'recent_activity',
+        },
+      ],
+    };
+  },
+  created () {
+    if (!this.$store.state.publicGuilds) this.fetchGuilds();
+  },
+  computed: {
+    guilds () {
+      return this.$store.state.publicGuilds;
+    },
+    filteredGuilds () {
+      let search = this.search;
+      let filters = this.filters;
+      let user = this.$store.state.user.data;
+      let filterGuild = this.filterGuild;
+      return this.guilds.filter((guild) => {
+        return filterGuild(guild, filters, search, user);
```

Comment:
this filtering shouldn't be done on the server given that we don't have all the guilds loaded?